balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -575,6 +607,12 @@ func (b *pickfirstBalancer) updateSubConnState(sd *scData, newState balancer.Sub
 		// the first address when the picker is used.
 		b.shutdownRemainingLocked(sd)
 		b.state = connectivity.Idle
+		// READY SubConn interspliced in between CONNECTING and IDLE, need to
+		// account for that.
+		if oldState == connectivity.Connecting && newState.ConnectivityState == connectivity.Idle {
```

Comment:
+1 to something like:

```go
if oldState == Connecting {
  // A known issue causes a race that prevents the READY state change notification.  This works around it.
```

Also it would be great if we could create an issue for that problem and link it here so we know to come back and delete this one it's resolved.

Reply:
Removed second conditional. Created a Github issue and linked to it.